### PR TITLE
Partial Cherry-pick of #886 and Additional Adjustments

### PR DESCRIPTION
**Pull request by kaituo**
### Description

This commit includes a partial cherry-pick of https://github.com/opensearch-project/anomaly-detection/pull/886 along with other necessary changes for compatibility with the 2.8 branch.

The cherry-picked changes from the anomaly-detection repository's main branch are not identical due to differences between the main and 2.8 branches. Specifically, the changes in https://github.com/opensearch-project/OpenSearch/pull/7165 present in the main branch are not incorporated in 2.8. To reconcile this, modifications related to ImmutableMap and ImmutableOpenMap were undone.

Additionally, only partial changes from https://github.com/opensearch-project/OpenSearch/commit/1e08b5a075fa6018be6c0af959b2d638c2743d56 were applied to 2.8. To address compiler failures, OpenSearchRejectedExecutionException was relocated from the org.opensearch.common.util.concurrent package to the org.opensearch.core.concurrency package.

I also incorporate selected changes from PR https://github.com/opensearch-project/anomaly-detection/pull/907 to enhance security in the 2.x branch. This update transitions from simpler passwords to a more secure, unguessable password system. The motivation behind this change is to address a potential security risk identified with simpler passwords (details at https://tinyurl.com/383em9zk). Without these updates, security tests would fail due to the vulnerability associated with simple passwords.

Tests completed:
- Successful Gradle build

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by jackiehanyang — DISMISSED** [review]

File: build.gradle [hunk 1]
```diff
@@ -733,8 +741,8 @@ dependencies {
     compileOnly "org.opensearch:opensearch-job-scheduler-spi:${job_scheduler_version}"
     implementation "org.opensearch:common-utils:${common_utils_version}"
     implementation "org.opensearch.client:opensearch-rest-client:${opensearch_version}"
-    implementation group: 'com.google.guava', name: 'guava', version:'31.0.1-jre'
-    implementation group: 'com.google.guava', name: 'failureaccess', version:'1.0.1'
+    compileOnly group: 'com.google.guava', name: 'guava', version:'31.0.1-jre'
```
Comment (amitgalitz) [inline build.gradle:744]: nit: these seemed to change in main in https://github.com/opensearch-project/anomaly-detection/pull/770. Do you know why this was never moved to 2.x?

**Comment by amitgalitz** [comment]
Should we make it standard to have separate limited PRs for any breaking change fixes in main to make backporting to 2.x easier?

File: build.gradle (hunk 1 above)
Comment (kaituo) [reply build.gradle:744]: not sure. Maybe due to my mistakes forgetting to add the changes to build.gradle when backporting.

**Comment by kaituo** [comment]
> Should we make it standard to have separate limited PRs for any breaking change fixes in main to make backporting to 2.x easier?

It is hard to enforce that standard. When the main branch is broken, I will try to change everything to make it compile and pass tests. It is not easy to add one PR for each core breaking changes. The core selectively backport from main to 2.8 also adds the complexity of the current PR.

**Review by ohltyler — APPROVED** [review]

**Review by amitgalitz — APPROVED** [review]

**Pull request merged**